### DISCUSSION TITLE
Running Ubuntu16 pybind output on Ubuntu 16 error

### DISCUSSION THREAD

**Discussion by Nir-Az** [pybind/pybind11#4351] [General]
Hi,

I built a python wrapper using Ubuntu18, when I try to run it on Ubuntu16 I get this error:
```
ImportError: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /home/administrator/work/python test/my_app.cpython-39-x86_64-linux-gnu.so)

```

Any idea how to make it work?
Thanks,
Nir

**Comment by jiwaszki** [comment]
@Nir-Az You are missing newer libs that were used to compile bindings, try to build Python3.9 from sources (I assume this from error message). However, this might be tricky process as Ubuntu16 has reached end-of-support stage.

You can also try updating the system to get newer GLIBC (I am not sure if 2.27 will be available for Ubuntu16):
```
apt-get update
apt-get install -y libc6 libc6-dev
```
There is also an option of getting them from sources, links:

- https://www.gnu.org/software/libc/sources.html
- https://ftp.gnu.org/gnu/glibc/

PS hello fellow Intel developer 👋

**Reply by Nir-Az** [reply]
Thanks for trying to help.

Apparently it's much more complicated than that..
```
administrator@ubuntu:~$ sudo apt-get install -y libc6 libc6-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libc6 is already the newest version (2.23-0ubuntu11.3).
libc6-dev is already the newest version (2.23-0ubuntu11.3).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```
as it seems the version you build pybind11 on is the minimal version you support..

**Reply by jiwaszki** [reply]
What about building glibc from sources? Is it solving a problem?

**Reply by Nir-Az** [reply]
According to what I read it is quite risky to do that because it can cause compatibility issues with other applications..
First I thought it's a pybins11 known  issue and I wanted to see if there is some easy solution.
But that's OK, we'll have to move to Ubuntu 18.
Thanks again for the support.
